src/frontend/src/session.rs

```diff
@@ -383,6 +383,40 @@ impl FrontendEnv {
         });
         join_handles.push(join_handle);
 
+        // Heartbeat
```

Comment:
There is another "heartbeat" between frontend and meta node at around L258.


```suggestion
        // Heartbeat between compute nodes (worker nodes) and frontend
```